assets/js/admin-settings/index.jsx

```diff
@@ -0,0 +1,128 @@
+/* global localizedData */
+import {
+  Button,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import React, { useState } from 'react';
+import useSiteOptions from '../services/hooks/use-site-options';
+import Rule from './rule';
+import { ruleCorral } from './styles';
+
+
+const AdminSettings = () => {
+  const [{ loading, setSettings, saving, settings }, saveSettings] = useSiteOptions();
+  const busy = loading || saving;
+  const [originIndex, setOriginIndex] = useState(null);
+  const [targetIndex, setTargetIndex] = useState(null);
+  const { apple_news_automation: ruleList } = settings;
+  const { fields } = wpLocalizedData;
+
+  /**
+   * Helper function for pushing to in-memory settings inside the useSiteOptions hook.
+   */
+   const sendSettings = (updatedRules) => {
+    const next = { ...settings, apple_news_automation: updatedRules };
+
+    // Enforce some defaults prior to save.
+    // Request will 500 when site_logo === null.
+    next.site_logo = next.site_logo ?? 0;
+
+    // Trigger state hook from useSiteOptions.
+    setSettings(next);
+  };
+
+  const addRule = () => {
+    const updatedRules = [...(ruleList ?? [])];
+    updatedRules.unshift({
+      field: '',
+      taxonomy: '',
+      term_id: 0,
+      value: '',
+    });
+    sendSettings(updatedRules);
```

Comment:
I'd recommend putting the new rule at the end rather than the beginning. This can also be simplified a bit.
```suggestion
    sendSettings([
      ...(ruleList ?? []),
      {
        field: '',
        taxonomy: '',
        term_id: 0,
        value: '',
      },
    ]);
```

Reply:
Only reservation here is that if the list of rules gets long enough a user might click to add a new rule and the rule is added out of view/offscreen at the bottom of the page.

Reply:
The add button should get moved to the end, so the rule would be added right above where the add button is. The general WP convention, and the convention used in the Apple News plugin elsewhere, is to have the save button at the bottom as well.